### ISSUE TITLE
Divine Shield Trait only applies it's damage reduction if it's in your Main-Hand.

### ISSUE THREAD

**Issue by SonicX8000** [TeamDman/PlusTiC#2]
This issue was originally opened on Oct 5, 2019 and it wasn't patched in newer versions. The issue still exists in the forked version after checking the code.

This is a copy/paste from the original post so while the listed mods are outdated, it still exists in their latest versions.

=====

Minecraft: 1.12.2
Forge: 14.23.5.2838

Mods:
armorplus-1.12.2-11.23.0.59
jei_1.12.2-4.15.0.279
Mantle-1.12-1.3.3.55
plustic-7.1.6.1
TConstruct-1.12.2-2.12.0.157
thedragonlib-1.12.2-5.3.0

The Divine Shield trait says that it grants Fire Resist as well as reducing the damage taken at the cost of durability, that is if it's in your Main-Hand.

If you have the trait in your Off-Hand... you will still get the Fire Resist potion effect however the damage isn't reduced but you also don't lose any durability. here are some tests with the trait while applying the 'instant_damage II' potion effect which damages me for 12 damage using an all Guardian Scale Broadsword.

Tests
If trait is in Main-Hand...
10 Damage Taken, 716/717 Durability.

If trait is in Off-Hand...
12 Damage Taken, 717/717 Durability.

Torohealth was used since it shows a Damage Particle.

P.S.
Is it possible to prevent the Fire Resist Particles from appearing if using the trait?

=====
Update for the issue:

In newer versions of ArmorPlus, they added support for Tinkers so the Guardian Scale parts will appear twice if you have PlusTic installed. You may need to disable the Guardian Scale config armor for ArmorPlus to avoid this.

One more thing to mention... while looking at the 'DivineShield.java' in the code it has this line here at 33...

`ItemStack tool = event.getEntityLiving().getHeldItemMainhand();`

...this is where it only checks if the trait is in the main-hand before it applies it's damage reduction. Could this be changed so it can check for the trait from either hand?

**Comment by TeamDman** [comment]
https://www.curseforge.com/minecraft/mc-mods/plusticminusbad/files/2945193
Please let me know if it does/does not work!

**Comment by SonicX8000** [comment]
Nope, still only reduces damage & grants Fire Resist if in the main-hand while the off-hand it does nothing besides granting Fire Resist.

I wonder if a Emuhand of some sorts could work?

**Comment by SonicX8000** [comment]
This bug still exists in v8.0.2 where it doesn't reduce damage if in the off-hand.

Also... I know that this is in legacy support but if bugfixes are still considered, the trait doesn't give Fire Resist in either hand if forced-loaded onto Artifacts from Tinker's Evolution. The code is perhaps checking if you have a valid material in the tool that has the trait in order to apply the potion effect however the damage reduction still works if in your main-hand.

Artifacts are Tinker's Tools/Armor that have modifiers or traits automatically applied on them as well as giving them extra modifiers, the drawback with them is that you can't modify them in any way unless you unseal them. You can find them as possible dungeon loot and you can also create your own artifacts and since this mod allows you to force-load traits... I gave some custom artifacts some of plustic's traits without needing the materials.

Light Metal trait also does this where it doesn't give the Attack & Mining Speed boost if loaded onto artifacts without the material that normally has them. It only applies the boost if you're using the material that has the trait.

**Comment by SonicX8000** [comment]
The above issues still exists with...

plustic-8.0.5

Divine Shield doesn't reduce damage if in your off-hand and if loaded as a modifier onto artifacts from Tinker's Evolution... Fire Resist won't apply from either hand.

Light Metal also won't boost mining speed if the trait is loaded as a modifier via artifacts. The boost only applies if it's on a tool material.
